sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -116,7 +116,7 @@ statement
         (TBLPROPERTIES tableProps=tablePropertyList))*
         (AS? query)?                                                   #createHiveTable
     | CREATE TABLE (IF NOT EXISTS)? target=tableIdentifier
-        LIKE source=tableIdentifier locationSpec?                      #createTableLike
+        LIKE source=tableIdentifier locationSpec? (tableProvider)?     #createTableLike
```

Comment:
Could we move `(tableProvider)? ` before `locationSpec?`? This is the order of how we create table:
https://github.com/apache/spark/blob/8bdcaa051f5e0e65e13cff3720326d7528f3600b/sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4#L99-L106

Reply:
cc @maropu

Reply:
Thanks Yuming, keep consistent with `createTableHeader `, it should move to the front of `locationSpec`.